includes/api/class-wc-admin-rest-reports-performance-indicators-controller.php

```diff
@@ -0,0 +1,268 @@
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $stat_data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		// Wrap the data in a response object.
+		$response = rest_ensure_response( $data );
+
+		$response->add_links( $this->prepare_links( $data ) );
+
+		/**
+		 * Filter a report returned from the API.
+		 *
+		 * Allows modification of the report data right before it is returned.
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param object           $report   The original report object.
+		 * @param WP_REST_Request  $request  Request used to generate the response.
+		 */
+		return apply_filters( 'woocommerce_rest_prepare_report_performance_indicators', $response, $stat_data, $request );
+	}
+
+	/**
+	 * Prepare links for the request.
+	 *
+	 * @param WC_Admin_Reports_Query $object Object data.
+	 * @return array
+	 */
+	protected function prepare_links( $object ) {
+		$pieces   = explode( '/', $object->stat );
+		$endpoint = $pieces[0];
+		$stat     = $pieces[1];
+
+		$links = array(
+			'report' => array(
+				'href' => rest_url( sprintf( '/%s/reports/%s/stats', $this->namespace, $endpoint ) ),
+			),
+		);
+
+		return $links;
+	}
+
+	/**
+	 * Get the Report's schema, conforming to JSON Schema.
+	 *
+	 * @return array
+	 */
+	public function get_item_schema() {
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'report_performance_indicator',
+			'type'       => 'object',
+			'properties' => array(
+				'stat'             => array(
+					'description' => __( 'Unique identifier for the resource.', 'wc-admin' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'value'      => array(
+					'description' => __( 'Value of the stat. Returns null if the stat does not exist or cannot be loaded.', 'wc-admin' ),
+					'type'        => 'number',
+					'context'     => array( 'view', 'edit' ),
+					'readonly'    => true,
+				),
+			),
+		);
+
+		return $this->add_additional_fields_schema( $schema );
+	}
+
+	/**
+	 * Get the query params for collections.
+	 *
+	 * @return array
+	 */
+	public function get_collection_params() {
+		$allowed_stats     = $this->get_allowed_stats();
+		if ( is_wp_error( $allowed_stats ) ) {
+			$allowed_stats = __( 'There was an issue loading the report endpoints', 'wc-admin' );
+		} else {
+			$allowed_stats = implode( ', ', $allowed_stats );
+		}
+
+		$params            = array();
+		$params['context'] = $this->get_context_param( array( 'default' => 'view' ) );
+		$params['stats']   = array(
+			'description'       => sprintf(
+				/* translators: Allowed values is a list of stat endpoints. */
+				__( 'Limit response to specific report stats. Allowed values: %s.', 'wc-admin' ),
+				$allowed_stats
```

Comment:
Ahh this is where it might be nice to make the result of `get_allowed_stats()` filterable - so if an `OPTIONS` request is sent to the endpoint, any extension-supported stat would be listed here.